crates/agglayer-storage/src/stores/local_network_state/mod.rs

```diff
@@ -0,0 +1,303 @@
+use std::{
+    collections::{BTreeMap, VecDeque},
+    path::Path,
+    sync::Arc,
+};
+
+use agglayer_types::{Hash, Keccak256Hasher, LocalNetworkStateData, NetworkId};
+use pessimistic_proof::{
+    local_balance_tree::LOCAL_BALANCE_TREE_DEPTH,
+    local_exit_tree::LocalExitTree,
+    nullifier_tree::NULLIFIER_TREE_DEPTH,
+    utils::smt::{Node, Smt},
+};
+
+use super::{
+    interfaces::reader::LocalNetworkStateReader, interfaces::writer::LocalNetworkStateWriter,
+};
+use crate::{
+    columns::{
+        balance_tree_per_network::BalanceTreePerNetworkColumn, local_exit_tree_per_network as LET,
+        local_exit_tree_per_network::LocalExitTreePerNetworkColumn,
+        nullifier_tree_per_network::NullifierTreePerNetworkColumn, ColumnSchema,
+    },
+    error::Error,
+    storage::DB,
+    types::{SmtKey, SmtKeyType, SmtValue},
+};
+
+#[cfg(test)]
+mod tests;
+
+/// A logical store for the local network states.
+pub struct LocalNetworkStateStore {
+    db: Arc<DB>,
+}
+
+impl LocalNetworkStateStore {
+    pub fn new(db: Arc<DB>) -> Self {
+        Self { db }
+    }
+
+    pub fn new_with_path(path: &Path) -> Result<Self, Error> {
+        let db = Arc::new(DB::open_cf(
+            path,
+            crate::storage::local_network_state_db_cf_definitions(),
+        )?);
+
+        Ok(Self { db })
+    }
+}
+
+impl LocalNetworkStateWriter for LocalNetworkStateStore {
+    fn write_local_network_state(
+        &self,
+        network_id: &NetworkId,
+        new_state: &LocalNetworkStateData,
+        new_leaves: &[Hash],
+    ) -> Result<(), Error> {
+        let network_id: u32 = (*network_id).into();
+
+        // Store the LET
+        {
+            let new_leaf_count = new_state.exit_tree.leaf_count;
+            let start_leaf_count = new_leaf_count - new_leaves.len() as u32;
+
+            if let Some(stored_exit_tree) = self.read_local_exit_tree(network_id.into())? {
+                if stored_exit_tree.leaf_count != start_leaf_count {
+                    return Err(Error::InconsistentState {
+                        network_id: network_id.into(),
+                    });
+                }
+            } else {
+                // state from scratch, nothing to do
+            }
+
+            // TODO: make batch write across the 3 cfs
```

Comment:
You can already use a `WriteBatch` I think

Reply:
done